internal/rbd/rbd_attach.go

```diff
@@ -193,27 +196,32 @@ func waitForPath(ctx context.Context, pool, namespace, image string, maxRetries
 	return "", false
 }
 
-// Check if rbd-nbd tools are installed.
-func checkRbdNbdTools() bool {
+// set features available with rbd-nbd, also return module loaded status.
+func setRbdNbdToolFeatures() bool {
 	// check if the module is loaded or compiled in
 	_, err := os.Stat(fmt.Sprintf("/sys/module/%s", moduleNbd))
 	if os.IsNotExist(err) {
 		// try to load the module
 		_, _, err = util.ExecCommand(context.TODO(), "modprobe", moduleNbd)
 		if err != nil {
-			log.ExtendedLogMsg("rbd-nbd: nbd modprobe failed with error %v", err)
+			log.WarningLogMsg("rbd-nbd: nbd modprobe failed with error %v", err)
 
-			return false
+			return false // failed to load nbd module
```

Comment:
no need to return bool from here? instead of that set `hasNBD` the same way you are handling the `hasNBDCookieSupport`variable

Reply:
Done now.